packages/concerto-core/test/introspect/scalardeclaration.js

```diff
@@ -304,3 +308,64 @@ describe('ScalarDeclaration', () => {
         });
     });
 });
+
+describe('ScalarDeclaration -  - Test for declarations using Import Aliasing', () => {
+
+    let modelManager;
+    let resolvedModelManager;
+
+    beforeEach(() => {
+        modelManager = new ModelManager({ strict: true, importAliasing: true, enableMapType: true});
+
+        const childModelCTO = fs.readFileSync(path.resolve(__dirname, '../data/aliasing/child.cto'), 'utf8');
+        const parentModelCTO = fs.readFileSync(path.resolve(__dirname, '../data/aliasing/parent.cto'), 'utf8');
+
+        modelManager.addCTOModel(childModelCTO, 'child@1.0.0.cto');
+        modelManager.addCTOModel(parentModelCTO, 'parent@1.0.0.cto');
+        const resolvedMetamodelChild = modelManager.resolveMetaModel(modelManager.getAst().models[0]);
+        const resolvedMetamodelParent = modelManager.resolveMetaModel(modelManager.getAst().models[1]);
+        resolvedModelManager = new ModelManager({ strict: true, importAliasing: true, enableMapType: true});
+        const resolvedModelFileChild = new ModelFile(resolvedModelManager, resolvedMetamodelChild, 'child@1.0.0.cto');
+        const resolvedModelFileParent = new ModelFile(resolvedModelManager, resolvedMetamodelParent, 'parent@1.0.0.cto');
+        resolvedModelManager.addModelFiles([resolvedModelFileChild, resolvedModelFileParent], ['child@1.0.0.cto', 'parent@1.0.0.cto']);
+    });
+
+    // describe.only('#getName', () => {
+    //     it('should return the scalar name', () => {
+    //         // This fails, current imports are not declared within a namespace. Is this a bug?
+    //         // Why are we treating imported declrations as foreign objects? Shouldn't they be treated as part of the namespace and declared?
+    //         // Extend this test to map keys and value as well
```

Comment:
@mttrbrts @dselman can y'all please review this comment?

Reply:
For the following model
```
namespace car@1.0.0

import person@1.0.0.Driver

concept Driver extends Driver {}
```
This will throw the error that `Drive cannot extend itself` instead of `Driver is already declared.`, we error on similar lines in JS classes.

Reply:
Also with respect to this test, getting the type's classdeclaration from model manager is a bit tricky because, `MyString` is in `child@1.0.0`.

And used in `parent@1.0.0.Child.str` where its aliased as `Str`. `getType` on the property only returns a string value as `Str`. Shouldn't it return the ClassDeclration object.

Reply:
What is the expected response in the test? and what is the actual value?

Reply:
```
namespace child@1.0.0
concept Child {
    o Integer age
}
scalar MyString extends String

```
```
@test(Kid)
namespace parent@1.0.0

import child@1.0.0.{MyString as Str, Child as Kid}

@test(Kid)
concept Child extends Kid {
   @test(Kid)
   o Kid kid
   o Str str
}

concept Student extends Child {
   o Str schoolName
   o Str name
   o Child myChild
}

@test(Kid)
map KidIndex {
   o Str
   @test(Kid)
   o Kid
}
```

From `parent.Child.str` I want to get the scalarDeclration object `Str`, on which I want to invoke the `getName`, `getNamspace` methods and test them.

Since `Str` is not an is imported, we never declare it in the namesapce, because of which we cannot fetch it scalarDeclration object.

Reply:
I'll log an issue to discuss this. This is unrelated to the changes here.